src/optimize.d

```diff
@@ -796,8 +796,23 @@ extern (C++) Expression Expression_optimize(Expression e, int result, bool keepL
                 return;
             }
             // If e2 *could* have been an integer, make it one.
-            if (e.e2.op == TOKfloat64 && (e.e2.toReal() == cast(sinteger_t)e.e2.toReal()))
-                e.e2 = new IntegerExp(e.loc, e.e2.toInteger(), Type.tint64);
+            if (e.e2.op == TOKfloat64)
+            {
+                version (all)
```

Comment:
I mean that is should either be checking `__VERSION__` instead of using `version(all)`, or at least have a comment saying that it can be removed once 2.068 is no longer supported.  The bugzilla link and description are great but having the version number explicitly there as well makes it obvious.